app/src/common/shared/org/mozilla/vrbrowser/browser/Accounts.kt

```diff
@@ -0,0 +1,277 @@
+                }
+            }
+        }
+
+        override fun onAuthenticationProblems() {
+            accountStatus = AccountStatus.NEEDS_RECONNECT
+            accountListeners.toMutableList().forEach {
+                Handler(Looper.getMainLooper()).post {
+                    it.onAuthenticationProblems()
+                }
+            }
+        }
+
+        override fun onLoggedOut() {
+            accountStatus = AccountStatus.SIGNED_OUT
+            accountListeners.toMutableList().forEach {
+                Handler(Looper.getMainLooper()).post {
+                    it.onLoggedOut()
+                }
+            }
+        }
+
+        override fun onProfileUpdated(profile: Profile) {
+            accountListeners.toMutableList().forEach {
+                Handler(Looper.getMainLooper()).post {
+                    it.onProfileUpdated(profile)
+                }
+            }
+        }
+    }
+
+    init {
+        services.accountManager.registerForSyncEvents(
+                syncStatusObserver, ProcessLifecycleOwner.get(), false
+        )
+        services.accountManager.register(accountObserver)
+        accountStatus = if (services.accountManager.authenticatedAccount() != null) {
+            if (services.accountManager.accountNeedsReauth()) {
+                AccountStatus.NEEDS_RECONNECT
+
+            } else {
+                AccountStatus.SIGNED_IN
+            }
+
+        } else {
+            AccountStatus.SIGNED_OUT
+        }
+    }
+
+
+    fun addAccountListener(aListener: AccountObserver) {
+        if (!accountListeners.contains(aListener)) {
+            accountListeners.add(aListener)
+        }
+    }
+
+    fun removeAccountListener(aListener: AccountObserver) {
+        accountListeners.remove(aListener)
+    }
+
+    fun removeAllAccountListeners() {
+        accountListeners.clear()
+    }
+
+    fun addSyncListener(aListener: SyncStatusObserver) {
+        if (!syncListeners.contains(aListener)) {
+            syncListeners.add(aListener)
+        }
+    }
+
+    fun removeSyncListener(aListener: SyncStatusObserver) {
+        syncListeners.remove(aListener)
+    }
+
+    fun removeAllSyncListeners() {
+        syncListeners.clear()
+    }
+
+    fun authUrlAsync(): CompletableFuture<String?>? {
+        return CoroutineScope(Dispatchers.Main).future {
+            services.accountManager.beginAuthenticationAsync().await()
+        }
+    }
+
+    fun refreshDevicesAsync(): CompletableFuture<Boolean?>? {
+        return CoroutineScope(Dispatchers.Main).future {
+            services.accountManager.authenticatedAccount()?.deviceConstellation()?.refreshDevicesAsync()?.await()
+        }
+    }
+
+    fun updateProfileAsync(): CompletableFuture<Unit?>? {
+        return CoroutineScope(Dispatchers.Main).future {
+            services.accountManager.updateProfileAsync().await()
+        }
+    }
+
+    fun syncNowAsync(reason: SyncReason = SyncReason.Startup,
```

Comment:
nit: `reason = SyncReason.User` is a better default, since that's what you pass in to this call in every instance except for engine status changes.

Reply:
Updated